server/src/Application/Services/Course/CourseService.cs

```diff
@@ -1,18 +1,21 @@
 ﻿using StudyZen.Application.Dtos;
 using StudyZen.Application.Repositories;
 using StudyZen.Domain.Entities;
+using FluentValidation;
 
 namespace StudyZen.Application.Services;
 
 public sealed class CourseService : ICourseService
 {
     private readonly ICourseRepository _courses;
     private readonly ILectureRepository _lectures;
+    private readonly IValidator<Course> _updatedCourseValidator;
 
-    public CourseService(ICourseRepository courses, ILectureRepository lectures)
+    public CourseService(ICourseRepository courses, ILectureRepository lectures, IValidator<Course> updatedCourseValidator)
```

Comment:
Instead of injecting each validator manually, we could inject a ValidationHandler, which takes the instance to validate and resolves the validator using GetService<IValidator<T>>() method of IServiceProvider